src/InstrumentationEngine/InstructionGraph.cpp

```diff
@@ -1140,23 +1104,9 @@ HRESULT MicrosoftInstrumentationEngine::CInstructionGraph::RemoveAll()
     while (pInstr != NULL)
     {
         pInstr->SetIsRemoved();
-
-        CInstruction* pPreviousInstruction = pInstr->PreviousInstructionInternal();
         CInstruction* pNextInstruction = pInstr->NextInstructionInternal();
-
-        if (pPreviousInstruction != NULL)
```

Comment:
The original code does seem redundant, but I'm curious did this cause issues? I'm trying to figure out what we were trying to do here originally.

Reply:
Yes, it did cause issues. The removal of the pointers on the previous instruction caused the last instruction to ref count down to zero before the last iteration, causing an AV. I don't know what the original purpose was of the additional removals.